operator/src/main/java/org/bf2/operator/managers/StrimziManager.java

```diff
@@ -146,39 +169,65 @@ private void updateStatus() {
     }
 
     /* test */ public void updateStrimziVersion(Deployment deployment) {
+        List<String> kafkaVersions = readKafkaVersions(deployment);
+        List<String> kafkaIbpVersions = kafkaVersions.stream().map(AbstractKafkaCluster::getKafkaIbpVersion).collect(Collectors.toCollection(ArrayList::new));
+        Map<String, String> relatedImages = readRelatedImagesAnnotation(deployment);
 
-        Optional<EnvVar> kafkaImagesEnvVar =
-                deployment.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv()
-                        .stream()
-                        .filter(ev -> ev.getName().equals("STRIMZI_KAFKA_IMAGES"))
-                        .findFirst();
-
-        List<String> kafkaVersions = Collections.emptyList();
-        List<String> kafkaIbpVersions = Collections.emptyList();
-        if (kafkaImagesEnvVar.isPresent() && kafkaImagesEnvVar.get().getValue() != null) {
-            String kafkaImages = kafkaImagesEnvVar.get().getValue();
-            String[] kafkaImagesList = kafkaImages.split("\n");
-            kafkaVersions = new ArrayList<>(kafkaImagesList.length);
-            kafkaIbpVersions = new ArrayList<>(kafkaImagesList.length);
-            for (String kafkaImage : kafkaImagesList) {
-                String kafkaVersion = kafkaImage.split("=")[0];
-                String kafkaIbpVersion = AbstractKafkaCluster.getKafkaIbpVersion(kafkaVersion);
-                kafkaVersions.add(kafkaVersion);
-                kafkaIbpVersions.add(kafkaIbpVersion);
-            }
-        }
+        StrimziVersionStatus strimziVersion = new StrimziVersionStatusBuilder()
+                .withVersion(deployment.getMetadata().getName())
+                .withKafkaVersions(kafkaVersions)
+                .withKafkaIbpVersions(kafkaIbpVersions)
+                .withReady(Readiness.isDeploymentReady(deployment))
+                .build();
 
         this.strimziVersions.put(deployment.getMetadata().getName(),
-                new StrimziVersionStatusBuilder()
-                        .withVersion(deployment.getMetadata().getName())
-                        .withKafkaVersions(kafkaVersions)
-                        .withKafkaIbpVersions(kafkaIbpVersions)
-                        .withReady(Readiness.isDeploymentReady(deployment))
-                        .build());
+                new ComponentVersions(strimziVersion, relatedImages));
+    }
+
+    List<String> readKafkaVersions(Deployment deployment) {
+        return readAnnotation(deployment, KAFKA_IMAGES_ANNOTATION)
+            .or(() -> readVersionEnvVar(deployment))
+            .map(this::parseKafkaVersions)
+            .orElseGet(Collections::emptyList);
+    }
+
+    Optional<String> readVersionEnvVar(Deployment deployment) {
+        return deployment.getSpec()
+                .getTemplate()
+                .getSpec()
+                .getContainers()
+                .get(0)
```

Comment:
Ideally this would use the container name.  It's possible for an init container or something else to be introduced down the road.